lib/Dialect/TTNN/Transforms/Passes.cpp

```diff
@@ -1064,4 +1068,94 @@ class TTNNCreateInputGenerators
   }
 };
 
+class TTNNModifySignaturesForDylib
+    : public impl::TTNNModifySignaturesForDylibBase<
+          TTNNModifySignaturesForDylib> {
+
+public:
+  using impl::TTNNModifySignaturesForDylibBase<
+      TTNNModifySignaturesForDylib>::TTNNModifySignaturesForDylibBase;
+
+  void runOnOperation() final {
+    ModuleOp module = getOperation();
+    IRRewriter rewriter(&getContext());
+
+    // Ensure that the module has a single region and a single block within that
+    // region
+    assert(module->getRegions().size() == 1);
+    assert(module->getRegion(0).getBlocks().size() == 1);
+
+    // Get the first block of the region at index 0
+    //
+    Block *firstBlock = module.getBody(0);
+
+    // Find all the func.func ops in the module that are "forward" functions
+    //
+    SmallVector<func::FuncOp, 1> forwardFuncOps;
+    for (mlir::Operation &op : firstBlock->getOperations()) {
+      if (mlir::func::FuncOp funcOp = dyn_cast<func::FuncOp>(op)) {
+
+        // Skip functions that are called elsewhere in the IR
```

Comment:
At this point we shouldn't have any utility functions in IR since @LPanosTT added inliner few months ago.  Given that asserting maybe has more sense here.

Reply:
I prefer not to assert because if we do, we're making this pass behave as a test for whether there's any utility functions - the pass will work (successfully) if there are utility functions. Thoughts?